runtime/appruntime/metrics/system/system.go

```diff
@@ -0,0 +1,19 @@
+package system
+
+import (
+	"runtime"
+)
+
+type SysMetrics struct {
+	MemoryUsageBytes uint64
+	NumGoRoutines    int
+}
+
+func ReadSysMetrics() SysMetrics {
+	var memStats runtime.MemStats
+	runtime.ReadMemStats(&memStats)
+	return SysMetrics{
+		MemoryUsageBytes: memStats.Alloc,
+		NumGoRoutines:    runtime.NumGoroutine(),
+	}
+}
```

Comment:
Given the repetition and chance of typo's I feel like this would be better if it returned something like:

```suggestion
func ReadSysMetrics() map[string]uint64 {
```

Where the key is the metric name `e_memory_usage` and the value is the current value.  (This works given all of them are gauges right now).

This means name's would be consistent, and if we add a new metric we only need to add it in one place

Reply:
Sounds good. I'll also add a list of constants to have the key names in one place.